control/autoware_autonomous_emergency_braking/README.md

```diff
@@ -0,0 +1,225 @@
+# Autonomous Emergency Braking (AEB)
+
+## Purpose / Role
+
+`autonomous_emergency_braking` is a module that prevents collisions with obstacles on the predicted path created by a control module or sensor values estimated from the control module.
+
+### Assumptions
+
+This module has following assumptions.
+
+- It is used when driving at low speeds (about 15 km/h).
+
+- The predicted path of the ego vehicle can be made from either the path created from sensors or the path created from a control module, or both.
+
+- The current speed and angular velocity can be obtained from the sensors of the ego vehicle, and it uses points as obstacles.
+
+- The AEBs target obstacles are 2D points that can be obtained from the input point cloud or by obtaining the intersection points between the predicted ego footprint path and a predicted object's shape.
+
+### IMU path generation: steering angle vs IMU's angular velocity
+
+Currently, the IMU-based path is generated using the angular velocity obtained by the IMU itself. It has been suggested that the steering angle could be used instead onf the angular velocity.
+
+The pros and cons of both approaches are:
+
+IMU angular velocity:
+
+- (+) Usually, it has high accuracy
+- (-)Vehicle vibration might introduce noise.
+
+Steering angle:
+
+- (+) Not so noisy
+- (-) May have a steering offset or a wrong gear ratio, and the steering angle of Autoware and the real steering may not be the same.
+
+For the moment, there are no plans to implement the steering angle on the path creation process of the AEB module.
+
+## Inner-workings / Algorithms
+
+AEB has the following steps before it outputs the emergency stop signal.
+
+1. Activate AEB if necessary.
+
+2. Generate a predicted path of the ego vehicle.
+
+3. Get target obstacles from the input point cloud and/or predicted object data.
+
+4. Estimate the closest obstacle speed.
+
+5. Collision check with target obstacles.
+
+6. Send emergency stop signals to `/diagnostics`.
+
+We give more details of each section below.
+
+### 1. Activate AEB if necessary
+
+We do not activate AEB module if it satisfies the following conditions.
+
+- Ego vehicle is not in autonomous driving state
+
+- When the ego vehicle is not moving (Current Velocity is very low)
```

Comment:
> Current Velocity is very low

It says that the flow velocity is very low, but please tell me the threshold value.

Also, where is it used in the source code?

Reply:
@sfukuta 
> the flow velocity threshold.

Sorry what do you mean by the flow velocity threshold?

> Also, where is it used in the source code?

checking in the following part
https://github.com/tier4/autoware.universe/pull/1528/files#diff-9e6b733808409255bcc5d8a574750015103ce298b94d7d54872ff348f6740d3eR418-R423
```
  // step2. create velocity data check if the vehicle stops or not
  constexpr double min_moving_velocity_th{0.1};
  const double current_v = current_velocity_ptr_->longitudinal_velocity;
  if (std::abs(current_v) < min_moving_velocity_th) {
    return false;
  }
 ```